packages/build-utils/src/transforms/README.md

```diff
@@ -0,0 +1,195 @@
+# Source file transforms
+
+This directory contains home-grown transforms for the build systems of the MetaMask applications.
+
+## Remove Fenced Code
+
+> `./remove-fenced-code.ts`
+
+### Usage
+
+Let's imagine you've added some fences to your source code.
+
+```typescript
+this.store.updateStructure({
+  /** ..., */
+  GasFeeController: this.gasFeeController,
+  TokenListController: this.tokenListController,
+  ///: BEGIN:ONLY_INCLUDE_IF(snaps)
+  SnapController: this.snapController,
+  ///: END:ONLY_INCLUDE_IF
+});
+```
+
+The transform should be applied on your raw source files as they are committed to
+your repository, before anything else (e.g. Babel, `tsc`, etc.) parses or modifies them.
+
+```typescript
+import {
+  FeatureLabels,
+  removeFencedCode,
+  lintTransformedFile,
+} from '@metamask/build-utils';
+
+// Let's imagine this function exists in your build system and is called immediately
+// after your source files are read from disk.
+async function applyTransforms(
+  filePath: string,
+  fileContent: string,
+  features: FeatureLabels,
+  shouldLintTransformedFiles: boolean,
```

Comment:
Although you're pointing `applyTransforms` as an example, should we highlight on the README a _default_ value for `shouldLintTransformedFiles` or it's optionality for sake of clarity for the readers?

Reply:
Done in 204d1426